front/components/vaults/EditVaultStaticDatasourcesViews.tsx

```diff
@@ -1,48 +1,63 @@
 import { Button, PlusIcon, Popup } from "@dust-tt/sparkle";
 import type {
   DataSourceType,
+  DataSourceViewWithConnectorType,
   PlanType,
   VaultType,
   WorkspaceType,
 } from "@dust-tt/types";
+import { isWebCrawlerConfiguration } from "@dust-tt/types";
 import { useRouter } from "next/router";
 import React, { useCallback, useState } from "react";
 
 import VaultFolderModal from "@app/components/vaults/VaultFolderModal";
 import VaultWebsiteModal from "@app/components/vaults/VaultWebsiteModal";
+import { useDataSourceViewConnectorConfiguration } from "@app/lib/swr/data_source_views";
 
-export function EditVaultStaticDataSourcesViews({
-  owner,
-  plan,
-  vault,
-  category,
-  dataSources,
-}: {
+interface EditVaultStaticDatasourcesViewsProps {
   owner: WorkspaceType;
+  isOpen: boolean;
+  setOpen: (isOpen: boolean) => void;
   plan: PlanType;
   vault: VaultType;
-  category: "folder" | "website";
   dataSources: DataSourceType[];
-}) {
+  dataSourceView: DataSourceViewWithConnectorType | null;
+  category: "folder" | "website";
+  onClose: () => void;
+}
+
+export function EditVaultStaticDatasourcesViews({
+  owner,
+  plan,
+  vault,
+  isOpen,
+  setOpen,
```

Comment:
one remaining :-) here we ahve both onClose and setOpen

Reply:
That one is kinda needed as it checks whether we the plan limits are reached. If they are then the modal doesn't open and we display a dialog. If limits aren't reached then we open the modal.